File: src/backend/app/Models/RoomBooking.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class RoomBooking extends Model
{
    protected $table = 'roomBooking';

    protected $primaryKey = 'roomBookingId';
    
    public $timestamps = false;

    protected $fillable = [
        'roomNumber',
        'bookingId',
        'cost',
        'fees'
    ];

    public function booking()
    {
        return $this->belongsTo(Booking::class);
    }

    public function Room()
    {
        return $this->belongsTo(Room::class);
    }
}

File: src/asp_net_backend/src/api/Models/RoomBooking.cs
namespace Restify.API.Models;

public class RoomBooking
{
	public DateOnly			StartDate { get; set; }
	public DateOnly			EndDate { get; set; }
	public decimal			Cost { get; set; }
	public decimal			Discount { get; set; }
	public decimal			Fees { get; set; }
	public BookingStatus	Status { get; set; }
	public uint				BookingId { get; set; }
	public uint				LodgingId { get; set; }
	public uint				Id { get; set; }
	public uint				RoomNumber { get; set; }

	public Booking	Booking { get; }
	public Room		Room { get; }
}
